csrc/id_model/loop_promotion.cpp

```diff
@@ -988,21 +1096,13 @@ std::unordered_map<ValGroup, IterDomain*> LoopPromotionMapBuilder::
           (int64_t)StmtSort::getExprsTo({loop_id->extent()}).size();
       auto this_is_const = loop_id->extent()->isConstInt();
 
-      // A group is allowed to have one single exact group of concrete
-      // IDs with a broadcast group.
-      if (promotion == nullptr ||
-          (promotion->isBroadcast() && !loop_id->isBroadcast())) {
+      if (promotion == nullptr) {
```

Comment:
Previously, a broadcast and non-broadcast mixed loop group was supported. No longer the case as the shortcut logic is scaled back.